R/chat.R

```diff
@@ -33,6 +33,10 @@ Chat <- R6::R6Class(
     initialize = function(provider, system_prompt = NULL, echo = "none") {
       private$provider <- provider
       private$echo <- echo
+      private$callbacks <- list(
+        tool_request = CallbackManager$new(),
+        tool_result = CallbackManager$new()
+      )
       self$set_system_prompt(system_prompt)
```

Comment:
```suggestion
      private$tool_request_callbacks <- CallbackManager$new(),
      private$tool_result_callbacks = CallbackManager$new()
      self$set_system_prompt(system_prompt)
```

IMO there's no much of a win with using a list here, even if you expect the number of managers to grow a bunch.